src/data/number/numberFn.ts

```diff
@@ -1,10 +1,90 @@
 import { assert, assertInteger } from '../../utils/assert'
-import { plString } from '../string/stringFn'
 import { PLNumber } from './PLNumber'
 
-export const plNumber = (value: number): PLNumber => new PLNumber(value)
+export interface DecimalResult {
+  intValue: number
+  decimals: number
+}
+
+export function plFloat(x: number): PLNumber {
+  return parseNumber(x.toString())
+}
+
+export function plNumber(intValue: number, decimals = 0): PLNumber {
+  assertInteger(intValue)
+  assertInteger(decimals)
+  return new PLNumber(intValue, decimals)
+}
+
+export function parseNumber(strValue: string): PLNumber {
+  assertNumeric(strValue)
+  const decimalObj = isScientific(strValue) ? parseScientificString(strValue) : parseDecimalString(strValue)
+  return plNumber(decimalObj.intValue, decimalObj.decimals)
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function assertNumeric(strValue: any): boolean {
+  return assert(strValue === '' || isNaN(strValue), `Invalid number: "${strValue}"`)
```

Comment:
string value never equel with `NaN`, if you really want to use this for check then you have to parse the value first:
```
isNaN(parseFloat(strValue))
```

Reply:
See my previous reply [here](https://github.com/maxinteger/pocket-lisp-stdlib/pull/14#discussion_r559078636).

Reply:
hmm I did not know about `isNaN` convert the passed value to Number. Good to know :D